presto-native-execution/presto_cpp/main/LinuxMemoryChecker.h

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <boost/regex.hpp>
+#include "presto_cpp/main/PeriodicMemoryChecker.h"
+
+namespace facebook::presto {
+class LinuxMemoryChecker;
+
+static LinuxMemoryChecker* linuxMemoryChecker = nullptr;
```

Comment:
This can go to the cpp file as well. This must be a singleton. So it has to be initialized with the object instead of a nullptr.